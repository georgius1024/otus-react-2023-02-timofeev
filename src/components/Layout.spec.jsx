import { render, screen, fireEvent } from "@testing-library/react";
import { vi } from "vitest";
import flushPromises from 'flush-promises'

import LayoutBuilder from "./Layout.tsx";

describe("LayoutBuilder", () => {

  it("renders without errors without user", () => {
    const Layout = LayoutBuilder(null, vi.fn(), vi.fn())
    expect(render(<Layout />)).toMatchSnapshot();
  });
  it("renders without errors with user", () => {
    const Layout = LayoutBuilder({id: 101}, vi.fn(), vi.fn())
    expect(render(<Layout />)).toMatchSnapshot();
  });
});
