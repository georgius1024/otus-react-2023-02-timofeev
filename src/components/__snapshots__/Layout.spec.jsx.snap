// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LayoutBuilder > renders without errors with user 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
      >
        <div
          class="section section-1"
        >
          <a>
            [LOGO]
          </a>
        </div>
        <div
          class="section section-2"
        >
          TODOAPP
        </div>
        <div
          class="section section-3"
        >
          <a>
            Logout
          </a>
        </div>
      </header>
      <main
        class="container"
      />
      <footer
        class="footer"
      >
        (c) Diligent student
      </footer>
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
    >
      <div
        class="section section-1"
      >
        <a>
          [LOGO]
        </a>
      </div>
      <div
        class="section section-2"
      >
        TODOAPP
      </div>
      <div
        class="section section-3"
      >
        <a>
          Logout
        </a>
      </div>
    </header>
    <main
      class="container"
    />
    <footer
      class="footer"
    >
      (c) Diligent student
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`LayoutBuilder > renders without errors without user 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
      >
        <div
          class="section section-1"
        >
          <a>
            [LOGO]
          </a>
        </div>
        <div
          class="section section-2"
        >
          TODOAPP
        </div>
        <div
          class="section section-3"
        >
          <a>
            Login
          </a>
           
          <a>
            Register
          </a>
        </div>
      </header>
      <main
        class="container"
      />
      <footer
        class="footer"
      >
        (c) Diligent student
      </footer>
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
    >
      <div
        class="section section-1"
      >
        <a>
          [LOGO]
        </a>
      </div>
      <div
        class="section section-2"
      >
        TODOAPP
      </div>
      <div
        class="section section-3"
      >
        <a>
          Login
        </a>
         
        <a>
          Register
        </a>
      </div>
    </header>
    <main
      class="container"
    />
    <footer
      class="footer"
    >
      (c) Diligent student
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
